lib/autoStopReporting.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+const pFinally = require('p-finally');
+const wrap = require('lodash.wrap');
+
+function autoStopReporting(compiler, stopReporting) {
+    compiler.run = wrap(compiler.run, (run, ...args) => {
+        try {
+            return pFinally(run(...args), stopReporting);
+        } catch (err) {
+            stopReporting();
+
+            throw err;
+        }
+    });
+
+    compiler.watch = wrap(compiler.watch, (watch, ...args) => {
+        try {
+            return watch(...args);
+        } catch (err) {
+            stopReporting();
+
+            throw err;
+        }
+    });
+
+    compiler.unwatch = wrap(compiler.unwatch, (unwatch, ...args) => pFinally(unwatch(...args), stopReporting));
+    compiler.resolve = wrap(compiler.resolve, (resolve, ...args) => pFinally(resolve(...args), stopReporting));
```

Comment:
We will probably remove this anyway, but calling resolve() should not stop reporting.